### PR TITLE
Fix profile credential loader known_hosts

### PR DESCRIPTION
**Pull request by Joerger**
The `HostKeyCallBack` logic used by the profile credential loader would only read the first CA certificate in a `known_hosts` file. Therefore, if you are/have logged into multiple clusters, the API client will fail to connect over ssh unless the first CA cert is the correct one.

This PR adds a loops over `ParseKnownHosts` to gather all CA certs as intended.

### CODE REVIEW

File: api/utils/sshutils/ssh.go [hunk 1]
```diff
@@ -174,12 +174,15 @@ func AsAgentKeys(sshCert *ssh.Certificate, privKey []byte) ([]agent.AddedKey, er
...
-		_, _, publicKey, _, _, err := ssh.ParseKnownHosts(caCert)
-		if err != nil {
-			return nil, trace.BadParameter("failed parsing CA cert: %v; raw CA cert line: %q", err, caCert)
+		var err error
+		for err == nil {
+			var publicKey ssh.PublicKey
+			if _, _, publicKey, _, caCert, err = ssh.ParseKnownHosts(caCert); err == nil {
+				trustedKeys = append(trustedKeys, publicKey)
+			}
 		}
```
Comment (andrejtokarcik) [inline api/utils/sshutils/ssh.go:183]: Parse errors shouldn't be ignored.
```suggestion
		}
		if err != io.EOF {
			return nil, trace.Wrap(err, "failed parsing CA cert: %q", caCert)
		}
```

File: api/utils/sshutils/ssh.go [hunk 2]
```diff
@@ -174,12 +175,17 @@ func AsAgentKeys(sshCert *ssh.Certificate, privKey []byte) ([]agent.AddedKey, er
...
-		if err != nil {
-			return nil, trace.BadParameter("failed parsing CA cert: %v; raw CA cert line: %q", err, caCert)
+		var err error
+		for err == nil {
+			var publicKey ssh.PublicKey
+			_, _, publicKey, _, caCert, err = ssh.ParseKnownHosts(caCert)
+			if err != nil && err != io.EOF {
+				return nil, trace.Wrap(err, "failed parsing CA cert: %v; raw CA cert line: %q", err, caCert)
+			}
+			trustedKeys = append(trustedKeys, publicKey)
```
Comment (andrejtokarcik) [inline api/utils/sshutils/ssh.go:185]: On a last iteration when `err == io.EOF` this would append a `nil` to `trustedKeys` which could be troublesome.

Comment (Joerger) [reply api/utils/sshutils/ssh.go:185]: :facepalm:

**Review by andrejtokarcik — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**